### ISSUE TITLE
False positive IDE0010 on exhaustive pattern matching in switch statement

### ISSUE THREAD

**Issue by OutsiderH** [dotnet/roslyn#84657]
**Version Used**: Visual Studio Community 2026 18.7.3

**Steps to Reproduce**:

1. Create new project
2. Add the sample code
``` c#
public class Test<T> where T : class {
    public void Foo() {
        T? obj = null;
        switch (obj) {
            case null:
                break;
            case not null:
                break;
        }
    }
}
```
Note that `T` can be replaced with any type that can be instantiated or any constrained generic type that allows `null` conversion.

**Diagnostic Id**:
IDE0010: 填充开关

**Expected Behavior**:
The analyzer should recognize that null and not null cover 100% of the input domain, and therefore IDE0010 should not be reported.

**Actual Behavior**:
The IDE0010 diagonolosis is reported.
When apply automatic fix (Add default case), `default:` case is added and CS0162 is reported, proving that the compiler already considers the switch statement exhaustive, whereas the IDE analyzer does not.

**Comment by CyrusNajmabadi** [comment]
pretty certain this is a dupe.  right now the compiler exposes no way for us to know that the switch is exhaustive.
